rosapi/src/rosapi/params.py

```diff
@@ -93,11 +93,10 @@ def delete_param(name, params_glob):
         return
     # If the glob list is empty (i.e. false) or the parameter matches
     # one of the glob strings, continue to delete the parameter.
-    with param_server_lock:
-        if has_param(name):
+    if has_param(name, params_glob):
+        with param_server_lock:
```

Comment:
I initially meant [this](https://github.com/RobotWebTools/rosbridge_suite/pull/284/files/3e1e949d483691f826495fafa63a6c6ffdf4fbda#diff-9b18b072495ba40106802093bb486fb5R87), but on closer inspection it's not calling `has_param(name)`, but rather `rospy.has_param(name)`.